### PR TITLE
improve example FromJSON instance

### PR DESCRIPTION
**Pull request by andrewthad**
Resolves https://github.com/bos/aeson/issues/510

### CODE REVIEW

File: Data/Aeson.hs [hunk 1]
```diff
@@ -274,11 +274,9 @@ eitherDecodeStrict' =
 -- > {-# LANGUAGE OverloadedStrings #-}
 -- >
 -- > instance FromJSON Person where
--- >     parseJSON (Object v) = Person <$>
--- >                            v .: "name" <*>
--- >                            v .: "age"
--- >     -- A non-Object value is of the wrong type, so fail.
--- >     parseJSON _          = empty
+-- >   parseJSON = withObject "Person" $ \v -> Person
```
Comment (phadej) [inline Data/Aeson.hs]: Most code examples use 4 space indents. It's worth to keep the consistent style.

**Comment by phadej** [comment]
Style issue, otherwise LGTM, thanks!

**Comment by RyanGlScott** [comment]
@andrewthad: There is at least one more comment in `aeson` that needs a similar fix. A quick search reveals [this one](https://github.com/bos/aeson/blob/d8dafaea947f50c890a1b7327a058b4d08359d91/Data/Aeson/Types/FromJSON.hs#L288).

**Comment by andrewthad** [comment]
@RyanGlScott Good catch. I'll get that fixed.

**Comment by andrewthad** [comment]
I ended up needing to leave the existing example since the docs use the existing example as a demonstration of `typeMismatch`. I've provided in the `withObject` approach as an alternative with the mention that it is "to be preferred". Let me know what you think.

**Comment by bergmark** [comment]
LGTM

**Pull request merged**